### ISSUE TITLE
[v2 FEATURE]: use early stopping instead of a fixed number of epochs

### ISSUE THREAD

**Issue by UnixJunkie** [chemprop/chemprop#848]
_**Is your feature request related to a problem? Please describe.**

It seems that the default model trains a DNN for a fixed number of iterations (20 or so?)

**Use-cases/examples of this new feature**

It might be better to use some early stopping criterion; e.g. the performance metric
on the validation set did not improve for the last 5 epochs, then stop training.

**Desired solution/workflow**

training a model might be faster than the current method

**Discussion**

N/A

**Additional context**

https://keras.io/api/callbacks/early_stopping/

**Comment by kevingreenman** [comment]
Hi @UnixJunkie, this feature is already implemented. The default number of epochs in v2 is 50 (can be changed with the `--epochs` option on the CLI), and you can do early stopping with the `--patience` CLI option.

**Comment by UnixJunkie** [comment]
Thanks, I'll try it, sorry for the noise.
